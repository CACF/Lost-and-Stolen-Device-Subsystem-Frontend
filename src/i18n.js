/*
Copyright (c) 2018 Qualcomm Technologies, Inc.
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted (subject to the limitations in the disclaimer below) provided that the following conditions are met:

* Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
* Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
* Neither the name of Qualcomm Technologies, Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
 
i18n
  .use(LanguageDetector)
  .init({
    // we init with resources
    resources: {
      en: {
        translations: {
          "welcomeApp": "Welcome to LSMS",
          "homeLink"  : "Home",
          "dashboardLink": "Dashboard",
          "newCaseLink": "New Case",
          "viewCaseLink": "View Case",
          "updateCaseLink": "Update Case",
          "searchCaseLink": "Search Cases",
          "allCasesLink"  : "All Cases",
          "pendingCaseLink": "Pending Cases",
          "blockedCaseLink": 'Blocked Cases',
          "recoveredCaseLink": 'Recovered Cases',
          "CaseStatusLink": "Case Status",
        }
      },
      es: {
        translations: {
          "welcomeApp": "Bienvenido a LSMS",
          "homeLink"  : "Casa",
          "dashboardLink": "Tablero",
          "newCaseLink"  : "Nuevo caso",
          "viewCaseLink": "Ver caso",
          "updateCaseLink": "Actualizar caso",
          "searchCaseLink"  : "Casos de b√∫squeda",
          "allCasesLink"  : "Todos los casos",
          "pendingCaseLink": "Casos pendientes",
          "blockedCaseLink": 'Casos bloqueados',
          "recoveredCaseLink": 'Casos recuperados',
          "CaseStatusLink": "Estado del caso",
        }
      },
      id: {
        translations: {
          "welcomeApp": "Selamat datang di LSMS",
          "homeLink"  : "Rumah",
          "dashboardLink": "Dasbor",
          "newCaseLink"  : "Kasus baru",
          "viewCaseLink": "Lihat Kasus",
          "updateCaseLink": "Perbarui Kasus",
          "searchCaseLink"  : "Kasus Pencarian",
          "allCasesLink"  : "Semua Kasus",
          "pendingCaseLink": "Kasus yang Tertunda",
          "blockedCaseLink": 'Kasus yang Diblokir',
          "recoveredCaseLink": 'Kasus yang Dipulihkan',
          "CaseStatusLink": "Status Kasus",
        }
      }
    },
    fallbackLng: 'en',
    debug: true,
 
    // have a common namespace used around the full app
    ns: ['translations'],
    defaultNS: 'translations',
 
    keySeparator: false, // we use content as keys
 
    interpolation: {
      escapeValue: false, // not needed for react!!
      formatSeparator: ','
    },
 
    react: {
      wait: true
    }
  });
 
export default i18n;