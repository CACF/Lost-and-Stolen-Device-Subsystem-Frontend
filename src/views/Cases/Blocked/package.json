{
  "name": "Blocked",
  "version": "0.0.0",
  "private": true,
  "main": "./Blocked.js"
}
