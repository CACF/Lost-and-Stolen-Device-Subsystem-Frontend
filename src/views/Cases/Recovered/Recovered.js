/*
Copyright (c) 2018 Qualcomm Technologies, Inc.
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted (subject to the limitations in the disclaimer below) provided that the following conditions are met:

* Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
* Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
* Neither the name of Qualcomm Technologies, Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

import React, { Component } from 'react';
import { translate, I18n } from 'react-i18next';
import {instance, errors, getAuthHeader} from './../../../utilities/helpers';
import CaseBox from '../../../components/CaseBox/CaseBox';
import BoxLoader from '../../../components/BoxLoader/BoxLoader';
import Pagination from "react-js-pagination";
import {PAGE_LIMIT, RECOVERED_CASE} from '../../../utilities/constants';
import {Card, CardHeader, Row, Col} from 'reactstrap';
import DataTableInfo from '../../../components/DataTable/DataTableInfo';

/**
 * This Stateful component generates a list of all Recovered Cases.
 * It uses some other components to fulfill its task, like CaseBox component, Pagination and loader component.
 */
class Recovered extends Component {
  constructor(props) {
    super(props);
    this.handlePageClick = this.handlePageClick.bind(this);
    this.getCasesFromServer = this.getCasesFromServer.bind(this);
    this.updateTokenHOC = this.updateTokenHOC.bind(this);

    this.state = {
      activePage: 1,
      data: null,
      loading: true,
      totalCases: 0,
      start: 1,
      limit: PAGE_LIMIT
    }
  }
  getCasesFromServer(config) {
      let start = this.state.start;
      let limit = this.state.limit;
      instance.get(`/cases?status=${RECOVERED_CASE}&start=${start}&limit=${limit}`, config)
          .then(response => {
              this.setState({ data: response.data, totalCases: (response.data || {}).count, loading: false});
          })
          .catch(error => {
              errors(this, error);
          })
  }

  updateTokenHOC(callingFunc) {
      let config = null;
      if(this.props.kc.isTokenExpired(0)) {
          this.props.kc.updateToken(0)
              .success(() => {
                  localStorage.setItem('token', this.props.kc.token)
                  config = {
                    headers: getAuthHeader(this.props.kc.token)
                  }
                  callingFunc(config);
              })
              .error(() => this.props.kc.logout());
      } else {
          config = {
            headers: getAuthHeader()
          }
          callingFunc(config);
      }
  }

  handlePageClick(page) {
    let a1 = 1;
    let d = this.state.limit;
   	let start = a1 + d * (page - 1);

	this.setState({start: start, activePage: page}, () => {
	  this.updateTokenHOC(this.getCasesFromServer);
	});
  }
  componentDidMount() {
    this.updateTokenHOC(this.getCasesFromServer);
  }
  render() {
    let recovered_cases = null;
    if(((this.state.data || {}).cases || []).length > 0) {
        recovered_cases = this.state.data.cases.map(recovered => (
            <CaseBox info={recovered} key={recovered.tracking_id}/>
        ));
    }
    return (
        <I18n ns="translations">
        {
          (t, { i18n }) => (
            <div className="animated fadeIn">
              <ul className="listbox">
                  {
                      (this.state.loading)
                          ?
                          (
                              <div>
                                  <BoxLoader />
                                  <BoxLoader />
                                  <BoxLoader />
                              </div>
                          )
                          : ((this.state.data || {}).cases || []).length > 0
                            ?  <div>
                                <Card className="mb-3">
                                    <CardHeader className="border-bottom-0">
                                        <b className="text-primary">{(this.state.totalCases > 1) ? `${this.state.totalCases} Recovered Cases found`: `${this.state.totalCases} Recovered Case found`}</b>
                                    </CardHeader>
                                </Card>
                                {recovered_cases}
                            </div>
                            : 'No Recovered Cases found'
                  }
              </ul>
              <Row>
                  <Col className='col-xs-12 col-xl-6'>
                    {(((this.state.data || {}).cases || []).length > 0 && this.state.totalCases > PAGE_LIMIT) &&
                      <div className='mt-3'>
                        <DataTableInfo start={this.state.start} limit={this.state.limit} total={this.state.totalCases} itemType={'cases'} />
                      </div>
                    }
                  </Col>
                {((((this.state.data || {}).cases || []).length > 0  && this.state.totalCases > PAGE_LIMIT) &&
                  <Col className='col-xs-12 col-xl-6'>
                  <Pagination
                      pageRangeDisplayed={window.matchMedia("(max-width: 575px)").matches ? 4 : 10}
                      activePage={this.state.activePage}
                      itemsCountPerPage={this.state.limit}
                      totalItemsCount={this.state.totalCases}
                      onChange={this.handlePageClick}
                      innerClass="pagination float-right mt-3"
                    /></Col>) || <div className="mb-3"></div>}
              </Row>
            </div>
          )
        }
        </I18n>
    )
  }
}

export default translate('translations')(Recovered);
