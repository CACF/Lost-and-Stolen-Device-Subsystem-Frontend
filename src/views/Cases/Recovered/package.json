{
  "name": "Recovered",
  "version": "0.0.0",
  "private": true,
  "main": "./Recovered.js"
}
