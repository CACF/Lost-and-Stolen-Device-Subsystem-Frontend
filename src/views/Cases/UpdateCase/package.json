{
  "name": "UpdateCase",
  "version": "0.0.0",
  "private": true,
  "main": "./UpdateCase.js"
}
