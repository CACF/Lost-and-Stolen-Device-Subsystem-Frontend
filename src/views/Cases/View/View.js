/*
Copyright (c) 2018 Qualcomm Technologies, Inc.
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted (subject to the limitations in the disclaimer below) provided that the following conditions are met:

* Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
* Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
* Neither the name of Qualcomm Technologies, Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

import React, { Component } from 'react';
import { Link } from "react-router-dom";
import { translate, I18n } from 'react-i18next';
import {instance, errors, getAuthHeader} from './../../../utilities/helpers';
import {RECOVERED_CASE, BLOCKED_CASE} from './../../../utilities/constants';
import { Button, Row, Col, Card, CardBody, CardHeader} from 'reactstrap';
import BoxLoader from '../../../components/BoxLoader/BoxLoader';

/**
 * This Stateful component provides View functionality of a Case.
 */
class View extends Component {
  constructor(props) {
    super(props);

    this.getCaseFromServer = this.getCaseFromServer.bind(this);
    this.updateTokenHOC = this.updateTokenHOC.bind(this);

    this.state = {
        data: null,
        loading: true
    }
  }
  componentDidMount() {
    this.updateTokenHOC(this.getCaseFromServer);
  }

  updateTokenHOC(callingFunc) {
      let config = null;
      if(this.props.kc.isTokenExpired(0)) {
          this.props.kc.updateToken(0)
              .success(() => {
                  localStorage.setItem('token', this.props.kc.token)
                  config = {
                    headers: getAuthHeader(this.props.kc.token)
                  }
                  callingFunc(config);
              })
              .error(() => this.props.kc.logout());
      } else {
          config = {
            headers: getAuthHeader()
          }
          callingFunc(config);
      }
  }

  getCaseFromServer() {
      const {tracking_id} = this.props.match.params;
      var config = {
        headers: getAuthHeader()
      }
      instance.get(`/case/${tracking_id}`, config)
          .then(response => {
              this.setState({ data: response.data, loading: false});
          })
          .catch(error => {
              errors(this, error);
          })
  }

  render() {
    let view_case = null;
    if(this.state.data) {
        const {tracking_id, updated_at, status, device_details, incident_details, personal_details, comments, creator} = this.state.data;
        const statusClass =
        (status === 'Pending')
        ?
            'text-primary'
        : (status === 'Recovered')
            ? 'text-success'
            : 'text-danger';
        view_case = <div key={tracking_id}>
          <Row >
            <Col xs="12" lg="8">
              <h4>Case Identifier: {tracking_id} - <small>Status <span className={statusClass}>{status}</span></small></h4>
              <p className="last-updated mb-0">Creator: <b>{creator.username}</b></p>
              <p className="last-updated">Last updated: <b>{updated_at}</b></p>
            </Col>
              <Col xs="12" lg="4">
                  {(status === 'Pending') ?
                      <div className="text-right pb-4">
                        <Link className="btn-sm btn btn-primary"
                            to={`/case-update/${ tracking_id }`}>Update</Link>{' '}
                        <Button color="success" size="sm" onClick={(e) => this.props.handleCaseStatus(e, tracking_id, RECOVERED_CASE)}>Recover</Button>{' '}
                        <Button color="danger" size="sm" onClick={(e) => this.props.handleCaseStatus(e, tracking_id, BLOCKED_CASE)}>Block</Button>
                      </div>
                      : (status === 'Blocked') ?
                          <div className="text-right pb-4">
                            <Button color="success" size="sm" onClick={(e) => this.props.handleCaseStatus(e, tracking_id, RECOVERED_CASE)}>Recover</Button>
                          </div>
                      : ''
                  }
              </Col>
          </Row>
          <Row>
            <Col xs="12">
                <Card>
                    <CardHeader>
                        <b>Device Description</b>
                    </CardHeader>
                    <CardBody>
                        <table className="table table-bordered table-sm mb-0">
                            <tbody>
                                <tr>
                                    <th>Brand</th>
                                    <td>{device_details.brand}</td>
                                </tr>
                                <tr>
                                    <th>Model name</th>
                                    <td>{device_details.model_name}</td>
                                </tr>
                                <tr>
                                    <th>Physical description</th>
                                    <td>{device_details.description}</td>
                                </tr>
                            </tbody>
                        </table>
                    </CardBody>
                </Card>
            </Col>
          </Row>
          <Row>
            <Col xs="12" sm="6">
                <Card>
                    <CardHeader>
                        <b>Affected MSISDN(s)</b>
                    </CardHeader>
                    <CardBody>
                        <table className="table table-bordered table-sm mb-0">
                            <tbody>
                            {device_details.msisdns.map((msisdn, index) => (
                               <tr key={index}>
                                    <td>{msisdn}</td>
                                </tr>
                            ))}
                            </tbody>
                        </table>
                    </CardBody>
                </Card>
            </Col>
            <Col xs="12" sm="6">
                <Card>
                    <CardHeader>
                        <b>Affected IMEI(s)</b>
                    </CardHeader>
                    <CardBody>
                        <table className="table table-bordered table-sm mb-0">
                            <tbody>
                            {device_details.imeis.map((imei, index) => (
                               <tr key={index}>
                                    <td>{imei}</td>
                                </tr>
                            ))}
                            </tbody>
                        </table>
                    </CardBody>
                </Card>
            </Col>
          </Row>
          <Row>
              <Col xs="12" md="6">
                  <Card>
                    <CardHeader>
                        <b>Incident Details</b>
                    </CardHeader>
                    <CardBody>
                        <table className="table table-bordered table-sm mb-0">
                            <tbody>
                                <tr>
                                    <th>Date of Incident</th>
                                    <td>{incident_details.incident_date}</td>
                                </tr>
                                <tr>
                                    <th>Nature of Incident</th>
                                    <td>{incident_details.incident_nature}</td>
                                </tr>
                            </tbody>
                        </table>
                    </CardBody>
                </Card>
              </Col>
              <Col xs="12" md="6">
                  <Card>
                    <CardHeader>
                        <b>Personal Details</b>
                    </CardHeader>
                    <CardBody>
                        <table className="table table-bordered table-sm mb-0">
                            <tbody>
                                <tr>
                                    <th>Full Name</th>
                                    <td>{personal_details.full_name}</td>
                                </tr>
                                <tr>
                                    <th>E-mail address</th>
                                    <td>{personal_details.email}</td>
                                </tr>
                                <tr>
                                    <th>Date of Birth</th>
                                    <td>{personal_details.dob}</td>
                                </tr>
                                <tr>
                                    <th>Govt. Identification No.</th>
                                    <td>{personal_details.gin}</td>
                                </tr>
                                <tr>
                                    <th>Alternate Phone Number</th>
                                    <td>{personal_details.number}</td>
                                </tr>
                                <tr>
                                    <th>Address</th>
                                    <td>{personal_details.address}</td>
                                </tr>
                            </tbody>
                        </table>
                    </CardBody>
                </Card>
              </Col>
          </Row>
          <Row>
              <Col xs="12">
                  <Card>
                    <CardHeader>
                        <b>Comments</b>
                    </CardHeader>
                    <CardBody className="p0">
                        {comments.length > 0 ? <article className="bglite">
                            {comments.map((comment, index) => (
                                <div className="comment-item" key={index}>
                                    <h6>{comment.username} :</h6>
                                    <div className="commentbox">
                                      <p className="comment-txt">{comment.comment}</p>
                                      <p className="comment-date"><span>{comment.comment_date}</span></p>
                                    </div>
                                </div>
                            ))}
                        </article> : (<p className="nodata"> No comments found</p>)
                        }
                    </CardBody>
                </Card>
              </Col>
          </Row>
        </div>
    }

    return (
        <I18n ns="translations">
        {
          (t, { i18n }) => (
            <div className="view-box animated fadeIn">
                <ul className="listbox">
                {
                    (this.state.loading)
                        ?
                        (
                            <div>
                                <BoxLoader />
                                <BoxLoader />
                                <BoxLoader />
                            </div>
                        )
                        : view_case
                }
                </ul>
            </div>
          )
        }
        </I18n>
    )
  }
}

export default translate('translations')(View);
