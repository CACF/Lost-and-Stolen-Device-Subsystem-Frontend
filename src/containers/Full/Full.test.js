import React from 'react';
import {shallow, mount} from 'enzyme';
import Full from './Full';
import {I18nextProvider} from "react-i18next";
import i18n from "./../../i18nTest";
import sinon from 'sinon';
import mockAxios from 'jest-mock-axios';
import { MemoryRouter } from 'react-router';
import NewCase from './../../views/NewCase/NewCase';
import SearchCases from "../../views/SearchCases";
import View from "../../views/Cases/View";
import UpdateCase from "../../views/Cases/UpdateCase";
import Pending from "../../views/Cases/Pending";
import Blocked from "../../views/Cases/Blocked";
import Recovered from "../../views/Cases/Recovered";
import {BrowserRouter as Router} from 'react-router-dom';

const mockKcProps = {
  'isTokenExpired': sinon.spy(),
  logout: sinon.spy()
}

const userDetails = {
  preferred_username: "User"
}

const match = { params: { tracking_id: 'IUJ34ET5' } }

const location = {
  pathname: '/new-case/'
};

const mockHeader = {
  "headers": {
    "Authorization": "Bearer null",
    "Content-Type": "application/json"
  }
}

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {
    },
    removeListener: () => {
    }
  })
});

const historyMock = { push: jest.fn() };

let responseObj = {
  data: {
        comments: [{
          comment: "Testing comment",
          comment_date: "2018-10-29T10:58:31.189731",
          user_id: "9a4403c2-5a48-4b79-9f30-02c0cc7799e0",
          username: "test"
        }],
        creator: {
          username: 'test',
          user_id: '9a4403c2-5a48-4b79-9f30-02c0cc7799e0'
        },
        device_details: {
          imeis: ['123456787655345'],
          description: 'test description',
          brand: 'test brand',
          model_name: 'test model',
          msisdns: ['03231234567']
        },
        incident_details: {
          incident_date: '2018-01-01',
          incident_nature: 'Test'
        },
        personal_details: {
          full_name: 'test',
          email: 'test@example.com',
          dob: null,
          gin: null,
          address: 'test address',
          number: null
        },
        status: 'Recovered',
        tracking_id: 'IUJ34ET5',
        updated_at: '2018-01-01 10:10:10',
        created_at: '2018-10-29 10:34:34'
      }
}

afterEach(() => {
  // cleaning up the mess left behind the previous test
  mockAxios.reset();
});

describe('Full component', () => {

  beforeEach(() => {
    // values stored in tests will also be available in other tests unless you run
    localStorage.clear();
    localStorage.userInfo = 'eyJzdWIiOiI5YTQ0MDNjMi01YTQ4LTRiNzktOWYzMC0wMmMwY2M3Nzk5ZTAiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJxdWFsY29tbSJ9';
  })

  test("if renders correctly", () => {
    const wrapper = shallow(<Full match={match}/>);
    expect(wrapper).toMatchSnapshot()
  });
  test('If Full component exists',()=>{
    const wrapper = shallow(<Full match={match}/>);
    expect(wrapper.exists()).toBe(true);
  })

  test('/new-case should redirect to New Case Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/new-case' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(NewCase)).toHaveLength(1);
  });

  test('/search-cases should redirect to Search Page', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/search-cases' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(SearchCases)).toHaveLength(1);
  });

  test('/case/{randomID} should redirect to View Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/case/IUJ34ET5' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(View)).toHaveLength(1);
  });

  test('/case-update/{randomID} should redirect to Update Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/case-update/IUJ34ET5' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(UpdateCase)).toHaveLength(1);
  });

  test('/cases/Pending should redirect to Pending Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/cases/pending' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(Pending)).toHaveLength(1);
  });

  test('/cases/blocked should redirect to Blocked Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/cases/blocked' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(Blocked)).toHaveLength(1);
  });

  test('/cases/recovered should redirect to Recovered Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/cases/recovered' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(Recovered)).toHaveLength(1);
  });

  test('/ should redirect to Search Page', () => {
    const mockLogout = sinon.spy();
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/' ]}>
        <I18nextProvider i18n={i18n}>
          <Full match={match} userDetails={userDetails} kc={mockKcProps} location={location} />
        </I18nextProvider>
      </MemoryRouter>
    );
    expect(wrapper.find(SearchCases)).toHaveLength(1);
  });

});
