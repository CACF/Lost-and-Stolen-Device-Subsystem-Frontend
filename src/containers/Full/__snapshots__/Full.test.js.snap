// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full component if renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Translate(Full)
    match={
      Object {
        "params": Object {
          "tracking_id": "IUJ34ET5",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bindI18n": "languageChanged loaded",
      "bindStore": "added removed",
      "children": [Function],
      "i18n": undefined,
      "match": Object {
        "params": Object {
          "tracking_id": "IUJ34ET5",
        },
      },
      "ns": Array [
        "translations",
      ],
      "nsMode": "default",
      "omitBoundRerender": true,
      "translateFuncName": "t",
      "usePureComponent": false,
      "wait": false,
      "withRef": false,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bindI18n": "languageChanged loaded",
        "bindStore": "added removed",
        "children": [Function],
        "i18n": undefined,
        "match": Object {
          "params": Object {
            "tracking_id": "IUJ34ET5",
          },
        },
        "ns": Array [
          "translations",
        ],
        "nsMode": "default",
        "omitBoundRerender": true,
        "translateFuncName": "t",
        "usePureComponent": false,
        "wait": false,
        "withRef": false,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
